recipes/openimageio/3.x/patches/3.0.0.3-cmake-targets.patch

```diff
@@ -0,0 +1,560 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index b204d5b7b..326b842b8 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -42,7 +42,7 @@ endif ()
+ 
+ # Set up module path for our own cmake modules and add some esential ones
+ list (APPEND CMAKE_MODULE_PATH
+-      "${PROJECT_SOURCE_DIR}/src/cmake/modules"
++      #"${PROJECT_SOURCE_DIR}/src/cmake/modules"
+       "${PROJECT_SOURCE_DIR}/src/cmake")
+ 
+ # Utilities
+@@ -233,7 +233,7 @@ if (OIIO_BUILD_TOOLS AND NOT BUILD_OIIOUTIL_ONLY)
+     add_subdirectory (src/iinfo)
+     add_subdirectory (src/maketx)
+     add_subdirectory (src/oiiotool)
+-    add_subdirectory (src/testtex)
++    #add_subdirectory (src/testtex)
+     add_subdirectory (src/iv)
+ endif ()
+ 
+diff --git src/cmake/add_oiio_plugin.cmake src/cmake/add_oiio_plugin.cmake
+index d65ac62c4..39d03a43b 100644
+--- src/cmake/add_oiio_plugin.cmake
++++ src/cmake/add_oiio_plugin.cmake
+@@ -72,7 +72,7 @@ macro (add_oiio_plugin)
+                                         OpenImageIO_EXPORTS)
+             target_include_directories (${_plugin_NAME} PRIVATE ${_plugin_INCLUDE_DIRS})
+             target_link_libraries (${_plugin_NAME} PUBLIC OpenImageIO
+-                                                   PRIVATE ${_plugin_LINK_LIBRARIES})
++                                                   PUBLIC ${_plugin_LINK_LIBRARIES})
+             set_target_properties (${_plugin_NAME} PROPERTIES PREFIX "" FOLDER "Plugins")
+             install_targets (${_plugin_NAME})
+         endif ()
+diff --git src/cmake/externalpackages.cmake src/cmake/externalpackages.cmake
+index f8670a83f..27c279166 100644
+--- src/cmake/externalpackages.cmake
++++ src/cmake/externalpackages.cmake
+@@ -66,23 +66,22 @@ checked_find_package (OpenEXR REQUIRED
+ # library.
+ include_directories(BEFORE ${IMATH_INCLUDES} ${OPENEXR_INCLUDES})
+ set (OPENIMAGEIO_IMATH_TARGETS Imath::Imath)
+-set (OPENIMAGEIO_OPENEXR_TARGETS OpenEXR::OpenEXR)
+-set (OPENIMAGEIO_IMATH_DEPENDENCY_VISIBILITY "PRIVATE" CACHE STRING
++set (OPENIMAGEIO_OPENEXR_TARGETS openexr::openexr)
```

Comment:
This can be set in the `CMakeDeps` generator

`cd.set_property("openexr", "cmake_target_name", "OpenEXR::OpenEXR")`
so that we can save some patch diffs

Sane with `OPENIMAGEIO_IMATH_DEPENDENCY_VISIBILITY`, I think it'd be better to set it in the `CMakeToolchain` generator

Reply:
Tried to change, somehow now it fails to link again. Not sure why. So I pushed the changes to see if it is a local problem or if the CI also runs into it.

Reply:
Fails locally, and the CI doesn't build for macOS, so maybe there is a problem. There is a load of the messages in the following style, happening for all the integrated tools when they are trying to link. Just a small excerpt:
```
[  0%] Built target CopyFiles
[ 13%] Built target OpenImageIO_Util
[ 88%] Built target OpenImageIO
[ 89%] Linking CXX executable ../../bin/iconvert
ld: warning: ignoring duplicate libraries: '-lc++', '/Users/ingmar/.conan2/p/b/opene614278b8c5e70/p/lib/libOpenEXRUtil-3_2.a'
Undefined symbols for architecture arm64:
  "Iex_3_3::IoExc::IoExc(char const*)", referenced from:
      OpenImageIO_v3_0::OpenEXRInput::valid_file(OpenImageIO_v3_0::Filesystem::IOProxy*) const in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec&, OpenImageIO_v3_0::ImageSpec const&) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInput::seek_subimage(int, int) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInputStream::seekg(unsigned long long) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInputStream::read(char*, int) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec const&, OpenImageIO_v3_0::ImageOutput::OpenMode) in libOpenImageIO.a[85](exroutput.cpp.o)
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, OpenImageIO_v3_0::ImageSpec const*) in libOpenImageIO.a[85](exroutput.cpp.o)
      ...
  "Iex_3_3::IoExc::~IoExc()", referenced from:
      OpenImageIO_v3_0::OpenEXRInput::valid_file(OpenImageIO_v3_0::Filesystem::IOProxy*) const in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec&, OpenImageIO_v3_0::ImageSpec const&) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInput::seek_subimage(int, int) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInputStream::seekg(unsigned long long) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXRInputStream::read(char*, int) in libOpenImageIO.a[84](exrinput.cpp.o)
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec const&, OpenImageIO_v3_0::ImageOutput::OpenMode) in libOpenImageIO.a[85](exroutput.cpp.o)
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, OpenImageIO_v3_0::ImageSpec const*) in libOpenImageIO.a[85](exroutput.cpp.o)
      ...
  "Imf_3_3::OutputFile::copyPixels(Imf_3_3::InputPart&)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::copy_image(OpenImageIO_v3_0::ImageInput*) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputFile::writePixels(int)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::write_scanlines(int, int, int, OpenImageIO_v3_0::TypeDesc, void const*, long long, long long) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputFile::setFrameBuffer(Imf_3_3::FrameBuffer const&)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::write_scanlines(int, int, int, OpenImageIO_v3_0::TypeDesc, void const*, long long, long long) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputFile::OutputFile(Imf_3_3::OStream&, Imf_3_3::Header const&, int)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec const&, OpenImageIO_v3_0::ImageOutput::OpenMode) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputPart::copyPixels(Imf_3_3::InputPart&)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::copy_image(OpenImageIO_v3_0::ImageInput*) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputPart::writePixels(int)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::write_scanlines(int, int, int, OpenImageIO_v3_0::TypeDesc, void const*, long long, long long) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputPart::setFrameBuffer(Imf_3_3::FrameBuffer const&)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::write_scanlines(int, int, int, OpenImageIO_v3_0::TypeDesc, void const*, long long, long long) in libOpenImageIO.a[85](exroutput.cpp.o)
  "Imf_3_3::OutputPart::OutputPart(Imf_3_3::MultiPartOutputFile&, int)", referenced from:
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, OpenImageIO_v3_0::ImageSpec const&, OpenImageIO_v3_0::ImageOutput::OpenMode) in libOpenImageIO.a[85](exroutput.cpp.o)
      OpenImageIO_v3_0::OpenEXROutput::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, OpenImageIO_v3_0::ImageSpec const*) in libOpenImageIO.a[85](exroutput.cpp.o)

```

Reply:
Will check tomorrow, thanks for the report!